### PR TITLE
feat: Feishu Sink supports multiple chats and more message types.

### LINKED ISSUE #99: support multiple chat group in Feishu Sink
support multiple chat group in Feishu Sink

### PR DESCRIPTION
**Pull request by wenfengwang**
**Changes:**
- support multiple chatgroups.
- support message type of post, image, share_chat, and interactive.
Close #99

### CODE REVIEW

**Comment by wenfengwang** [comment]
close #99

File: connectors/sink-feishu/internal/bot.go [hunk 1]
```diff
@@ -18,30 +18,216 @@ import (
...
+}
+
+type WebHook struct {
+	Address   string `json:"address" yaml:"address"`
+	Signature string `json:"signature" yaml:"signature"`
+	ChatGroup string `json:"chat_group" yaml:"chat_group"`
+}
+
+func (wh WebHook) Validate() error {
+	if wh.Address == "" {
```
Comment (xdlbdy) [inline connectors/sink-feishu/internal/bot.go:80]: It is no problem, but you also can add the tag `validate:"required"`

File: connectors/sink-feishu/internal/server.go [hunk 2]
```diff
@@ -17,13 +17,12 @@ package internal
 import (
 	"context"
 	"fmt"
+	"github.com/go-resty/resty/v2"
```
Comment (xdlbdy) [inline connectors/sink-feishu/internal/server.go:20]: fmt

File: connectors/sink-feishu/internal/server.go [hunk 3]
```diff
@@ -147,9 +134,11 @@ func (f *feishuSink) Initialize(_ context.Context, cfg cdkgo.ConfigAccessor) err
 	if !ok {
 		return nil
 	}
+	if err := _cfg.Validate(); err != nil {
```
Comment (xdlbdy) [inline connectors/sink-feishu/internal/server.go:137]: no need, because cdk has called this method

File: connectors/sink-feishu/internal/bot.go [hunk 4]
```diff
@@ -18,30 +18,216 @@ import (
...
+	}
+	return b.processResponse(e, res)
+}
+
+func (b *bot) sendImageMessage(e *v2.Event, wh WebHook) error {
+	content := map[string]interface{}{
+		"image_key": string(e.Data()),
+	}
+
+	res, err := b.httpClient.R().SetBody(b.generatePayload(content, imageMessage, wh)).Post(wh.Address)
```
Comment (xdlbdy) [inline connectors/sink-feishu/internal/bot.go:187]: too much repeated

File: connectors/sink-feishu/internal/server.go [hunk 5]
```diff
@@ -36,38 +35,29 @@ const (
...
-		return errors.New("feishu: bot webhook can't be empty")
-	}
-	return nil
 }
 
 type feishuConfig struct {
 	cdkgo.SinkConfig `json:",inline" yaml:",inline"`
 	Enable           []string  `json:"enable" yaml:"enable"`
 	Bot              BotConfig `json:"bot" yaml:"bot"`
 	Secret           Secret    `json:"secret" yaml:"secret"`
```
Comment (xdlbdy) [inline connectors/sink-feishu/internal/server.go:53]: secret struct no used ,remove it

File: connectors/sink-feishu/README.md [hunk 6]
```diff
@@ -149,11 +294,13 @@ data:
...
-      # write right webook
-      webhook: "https://open.feishu.cn/open-apis/bot/v2/hook/xxxxxx"
-    secret:
-      # write right bot_signature
-      bot_signature: "xxxxx"
+      webhooks:
+        - chat_group: "bot1"
+          signature: "xxxxxxxxxx"
+          address: "https://open.feishu.cn/open-apis/bot/v2/hook/xxxxxxxxxx"
+        - chat_group: "bot1"
```
Comment (xdlbdy) [inline connectors/sink-feishu/README.md:301]:
```suggestion
        - chat_group: "bot2"
```

File: connectors/sink-feishu/README.md [hunk 7]
```diff
@@ -97,32 +99,175 @@ now, you cloud see a notice in your chat group.
...
+## How to use
 
+### Configuration
 The default path is `/vance/config/config.yml`. if you want to change the default path, you can set env `CONNECTOR_CONFIG` to
 tell Feishu Sink.
 
+| Name                 | Required | Default | Description                                  |
+|:---------------------|:--------:|:-------:|----------------------------------------------|
+| enable               | **YES**  |    -    | service list you want Feishu Sink is enabled |
+| bot.webhooks         | **YES**  |    -    | list of chat-group's configuration           |
```
Comment (xdlbdy) [inline connectors/sink-feishu/README.md:111]: add chat_group, signature, and address description

File: connectors/sink-feishu/internal/bot.go [hunk 8]
```diff
@@ -18,30 +18,216 @@ import (
...
+	"github.com/linkall-labs/cdk-go/log"
+	"github.com/tidwall/gjson"
 )
 
-func (f *feishuSink) sendTextToFeishuBot(e *v2.Event) error {
-	t := time.Now()
+type messageType string
+
+const (
+	xChatGroupID = "xvfeishuchatgroup"
```
Comment (xdlbdy) [inline connectors/sink-feishu/internal/bot.go:36]: this is feishu sink, is it cant be modified to "xvchatgroup"

File: connectors/sink-feishu/internal/bot.go (hunk 1 above)
Comment (wenfengwang) [reply connectors/sink-feishu/internal/bot.go:80]: It doesn't work in my local

File: connectors/sink-feishu/internal/bot.go (hunk 4 above)
Comment (wenfengwang) [reply connectors/sink-feishu/internal/bot.go:187]: I can't get you meaning.

File: connectors/sink-feishu/internal/bot.go (hunk 8 above)
Comment (wenfengwang) [reply connectors/sink-feishu/internal/bot.go:36]: The reason I added the prefix aims to clarify when users write subscriptions (users may use the different sink in subscription.)

Comment (xdlbdy) [reply connectors/sink-feishu/internal/bot.go:36]: maybe you are right, the problem needs to be resolved by vanus but not by sink

File: connectors/sink-feishu/internal/bot.go (hunk 1 above)
Comment (xdlbdy) [reply connectors/sink-feishu/internal/bot.go:80]: ok, I will test it in the cdk too.

File: connectors/sink-feishu/internal/bot.go (hunk 4 above)
Comment (xdlbdy) [reply connectors/sink-feishu/internal/bot.go:187]: sorry, I didn't get a good look

File: connectors/sink-feishu/internal/server.go [hunk 9]
```diff
@@ -78,16 +68,11 @@ func (fc *feishuConfig) Validate() error {
...
 	}
-
-	if fc.Secret.BotSignature == "" {
-		return errors.New("feishu: secret can't be empty")
-	}
-
 	return nil
 }
 
 func (fc *feishuConfig) GetSecret() cdkgo.SecretAccessor {
```
Comment (xdlbdy) [inline connectors/sink-feishu/internal/server.go:74]: you can remove it which is same to the default implement

File: connectors/sink-feishu/internal/server.go [hunk 10]
```diff
@@ -17,13 +17,13 @@ package internal
 import (
 	"context"
 	"fmt"
+	"github.com/pkg/errors"
```
Comment (xdlbdy) [inline connectors/sink-feishu/internal/server.go:20]: fmt

**Pull request merged**